projects/ui/src/Utility/accessTokenRequest.ts

```diff
@@ -0,0 +1,61 @@
+export const doAccessTokenRequest = async (
+  formData: Record<string, string>,
+  grantType: "password" | "refresh_token",
+  tokenEndpoint: string,
+  clientId: string,
+  clientSecret: string
+) => {
+  //
+  // Build the request payload for a new oauth access token.
+  //
+  formData.grant_type = grantType;
+  formData.client_id = clientId;
+  formData.client_secret = clientSecret;
+  if (
+    (!(formData.username ?? "").trim() || !(formData.password ?? "").trim()) &&
+    !formData.refresh_token
+  ) {
+    return undefined;
+  }
+  //
+  // Encode the form request payload.
+  //
+  const payloadItems: string[] = [];
+  for (const property in formData) {
+    if (!formData.hasOwnProperty(property)) continue;
+    const encodedKey = encodeURIComponent(property);
+    const encodedValue = encodeURIComponent(formData[property]);
+    payloadItems.push(`${encodedKey}=${encodedValue}`);
+  }
+  //
+  // Make the request
+  //
+  const rawRes = await fetch(tokenEndpoint, {
+    headers: {
+      "Content-Type": "application/x-www-form-urlencoded;charset=UTF-8",
+    },
+    method: "POST",
+    body: payloadItems.join("&"),
```

Comment:
nit: I could be wrong (it might be a server/header setting), but I think JS `fetch` can accept a stringified json object as a body and handle it properly

Reply:
Since this is `application/x-www-form-urlencoded`, the body payload had to be formatted differently. I used this for reference: https://stackoverflow.com/a/37562814.

Reply:
Ah, alright! That's a shame; good to know though